Tools/autotest/arduplane.py

```diff
@@ -3720,6 +3720,35 @@ def FenceAutoEnableDisableSwitch(self):
         if (not (m.onboard_control_sensors_health & fence_bit)):
             raise NotAchievedException("Fence breach not cleared")
 
+        self.progress("Fly above ceiling and check there is a breach and switch to AUTOLAND mode")
+        self.set_parameters({
+            "FENCE_ACTION": 8,   # Set action to AUTOLAND if possible
+            })
+
+        self.set_rc(3, 2000)
+        self.set_rc(2, 1000)
+
+        self.wait_statustext("Max Alt fence breached", timeout=10, check_context=True)
+        self.wait_mode(26) # AUTOLAND,need pymavlink .43 to use text name
+        self.assert_fence_sys_status(True, True, False)
+
+        self.set_rc(3, 1500)
+        self.set_rc(2, 1500)
+
+        self.progress("Wait for cruise alt reached")
+        self.wait_altitude(cruise_alt-5, cruise_alt+5, relative=True, timeout=30)
```

Comment:
```suggestion
        self.wait_altitude(
            cruise_alt-5,
            cruise_alt+5,
            relative=True,
            timeout=30,
            minimum_duration=5,
        )
```

`minimum_duration` makes sure you're not just passing through